packages/express-did-auth/test/setup-app.cookies.integration.test.ts

```diff
@@ -1,118 +1,145 @@
+    response = await agent.post('/refresh-token')
+      .set('Cookie', `${secret}; ${removeExtraCookieAttributes(tokens[0])}; ${removeExtraCookieAttributes(tokens[1])}`)
+      .set(CSRF_TOKEN_HEADER_NAME, csrfToken)
+      .expect(200)
 
-    oldRefreshTokenCookie = refreshTokenCookie;
-    [accessTokenCookie, refreshTokenCookie] = cookies
+    // save old tokens to compare then
+    const oldTokens = tokens
 
-    MockDate.reset()
-  })
+    tokens = response.headers['set-cookie']
+    expect(tokens).toHaveLength(2)
+    expect(tokens[0]).toContain('authorization')
+    expect(tokens[1]).toContain('refresh-token')
+
+    // no tokens in the body
+    expect(response.body).toEqual({})
 
-  it('5b. POST /refresh-token with old one should fail', async () => {
-    const response = await cookieAgent.post('/refresh-token').set('Cookie', oldRefreshTokenCookie).expect(401)
+    // new tokens must be different
+    expect(tokens).not.toEqual(oldTokens)
+    expect(tokens[0]).not.toEqual(oldTokens[0])
+    expect(tokens[1]).not.toEqual(oldTokens[1])
+
+    // 5b. POST /refresh-token with old one should fail
+    response = await agent.post('/refresh-token')
+      .set('Cookie', `${secret}; ${removeExtraCookieAttributes(oldTokens[0])}; ${removeExtraCookieAttributes(oldTokens[1])}`)
+      .set(CSRF_TOKEN_HEADER_NAME, csrfToken)
+      .expect(401)
 
     expect(response.text).toEqual(INVALID_OR_EXPIRED_SESSION)
-  })
 
-  it('6. POST /logout with no access token should fail', async () => {
-    const response = await cookieAgent.post('/logout').expect(401)
+    // 6. POST /logout with no access token should fail
+    response = await agent.post('/logout')
+      .set('Cookie', secret)
+      .set(CSRF_TOKEN_HEADER_NAME, csrfToken)
+      .expect(401)
 
     expect(response.text).toEqual(NO_ACCESS_TOKEN)
-  })
 
-  it('6b. POST /logout with proper access token', async () => {
-    MockDate.set(Date.now() + 20000) // move the clock after the time set in 5. POST /refresh-token'
-
-    await cookieAgent
-      .post('/logout')
-      .set('Cookie', accessTokenCookie)
+    // 6b. POST /logout with proper access token
+    response = await agent.post('/logout')
+      .set('Cookie', `${secret}; ${removeExtraCookieAttributes(tokens[0])}; ${removeExtraCookieAttributes(tokens[1])}`)
+      .set(CSRF_TOKEN_HEADER_NAME, csrfToken)
       .expect(200)
 
-    MockDate.reset()
-  })
-
-  it('7. POST /refresh-token with logged out session one should fail', async () => {
-    const response = await cookieAgent.post('/refresh-token').set('Cookie', refreshTokenCookie).expect(401)
+    // 7. POST /refresh-token with logged out session one should fail
+    response = await agent.post('/refresh-token')
+      .set('Cookie', `${secret}; ${removeExtraCookieAttributes(tokens[0])}; ${removeExtraCookieAttributes(tokens[1])}`)
+      .set(CSRF_TOKEN_HEADER_NAME, csrfToken)
+      .expect(401)
 
     expect(response.text).toEqual(INVALID_OR_EXPIRED_SESSION)
   })
```

Comment:
I would add tow extra test here... Send a request with the csrf header, but without the cookie. it should fail

Reply:
Is pretty hard to understand to me, but it seems that the secret cookie is automatically added by `supertest`, the testing framework we are using. It is really weird, because it only adds the `_csrf` cookie and not the other ones that I generate like `authorization` or `refresh-token`. I have to add them manually